### ISSUE TITLE
Add Environment to add a Prefix for reverse-proxies

### ISSUE THREAD

**Issue by poldi1405** [metroline/metroline#15]
It just so happens that I hit the next wall:

When trying to set up Metroline behind a reverse proxy that does not map it to it's root directory, the possibility to set a prefix would be great, as metroline just assumes it runs in the root directory. I want it in the "directory" `ci` though. My nginx config looks like this, but I've hit a wall: 2.bcd1dd83.chunk.js requests the .env file which it suspects to be at the root directory.

```conf
server {
	listen 80;

	location / {
		proxy_pass http://gitea:3000/;
	}

	location /ci-backend/ {
		proxy_pass http://ciserver/;
	}

	location /ci/ {
		proxy_pass http://ciui/;
	}

	location /static/ {
		proxy_pass http://ciui/static/;
	}
}
```

Sure, I can setup routes for every file that is requested but that would be just not "elegant" but also a bit dumb imo.

**Comment by gempain** [comment]
@poldi1405 sorry for the delay and the inconvenience ! 

When we deploy to production, we generally use [Traefik](https://traefik.io/) or [Caddy](https://caddyserver.com/) and deploy our apps to multiple subdomains to avoid path conflicts. We recommend against trying to deploy multiple apps on sub-paths, because those paths could conflict with your `Gitea` installation (to what I can read from your example file).

But anyhow, what matters is that we help you with the setup you have, so let's try to find a solution - if we can. The right person to answer this would be @pimartin, he's the deployment expert in this project. He'll be back tomorrow morning and can help you with that !

**Comment by poldi1405** [comment]
In general the issue of collisions would be a real problem, but as i am controlling the Gitea instance I can ensure that there is no user called ci or ci-backend. Then the issue of collisions would be resolved.

**Comment by gempain** [comment]
Yes makes sense. So after a bit search, I found that we can set [`basename`](https://reactrouter.com/web/api/BrowserRouter/basename-string) on the router, but that won't fix the `/static` issue. For that, React [docs](https://create-react-app.dev/docs/deployment/#building-for-relative-paths) mention that we could set a `homepage` property in the `package.json` to the exact URL where your app is served (which won't work in this case because it's per-deployment specific), or setting `homepage` to `"."` which would allow serving the app from [any location](https://create-react-app.dev/docs/deployment/#serving-the-same-build-from-different-paths). However, that second solution mentions that **If you are not using the HTML5 pushState history API or not using client-side routing at all, it is unnecessary to specify the URL from which your app will be served.**, so I don't this would work here as we do use routing, but I'll try to test when I get to the office.

The main problem is that those things happen at build time, and I don't think an environment variable would work. This is something you might have to do on your own: clone, update those properties, and build your specific version. We could set the `basename` property as a build environment variable so that your custom build works properly. The build script could be shared in this repository for others that would like the same type of deployment as you.

But again, IMHO, using subdomains is so simple that I would just advise to deploy that way, it'll make your life simple and will probably be easier to remember as well (in terms of where your apps are), and you're free from unexpected collisions as well. That said, it's just, again, my humble opinion.

**Comment by pimartin** [comment]
@gempain I've used the `.` base href in Angular apps in other projects. I suspect it could work here too.

**Comment by gempain** [comment]
@poldi1405 haven't forgotten about you, I have busy at https://github.com/getmeli/meli as you may have seen is trending on GitHub
